### PR TITLE
Nicholasgodfreyclarke/log message bytes from cli

### PR DESCRIPTION
**Pull request by nicholasgodfreyclarke**
#### Problem

Currently the maximum size of program logs is hardcoded to 10K bytes https://github.com/solana-labs/solana/blob/118c1e9e4855b1edf315cbcfa2a513f4ceaaf837/program-runtime/src/log_collector.rs#L4

In some cases the program logs can be truncated - [example here](https://explorer.solana.com/tx/4nHwgUJshL8x9wnq1yUP8YhBcuuDo29GmyF3YZabCgPr4e3mvMTbLvQfG1kZdTpesRvaXug3cfNoqWo7yCf5qUgg). Particularly in the case where a transaction contains many instructions.

Beyond losing general logs - this causes particular issues when we rely on anchor events to be serialized to the program logs for analytics. Lost events means lost data.

[Related github issue](https://github.com/solana-labs/solana/issues/22301)

#### Summary of Changes

This PR adds a new CLI argument `log_messages_bytes_limit` to solana-validator and solana-test-validator that allows one to override the default 10k limit.
If the argument is not set - it reverts to the 10k limit.

### CODE REVIEW

**Comment by nicholasgodfreyclarke** [comment]
Hi @willhickey, linking you here since I see you self-assigned this issue in the linked issue above.

**Review by willhickey — COMMENTED** [review]
Thanks for the PR!

Just one request: can you add the new arg to [bootstrap-validator.sh](https://github.com/solana-labs/solana/blob/master/multinode-demo/bootstrap-validator.sh)

I think this will work:
```
    elif [[ $1 = --log-messages-bytes-limit ]]; then
      args+=("$1" "$2")
      shift 2
```

Edit: Looks like my attempt to resolve conflicts left something broken. I have to step away from my desk for a bit but I'll have a look later this evening.

**Review by willhickey — APPROVED** [review]
Looks good. Thanks!

**Comment by nicholasgodfreyclarke** [comment]
> Looks good. Thanks!

Thanks for the review!

File: banking-bench/src/main.rs [hunk 1]
```diff
@@ -213,6 +213,12 @@ fn main() {
                 .takes_value(true)
                 .help("Number of threads to use in the banking stage"),
         )
+        .arg(
+            Arg::new("log_messages_bytes_limit")
+                .long("log-messages-bytes-limit")
+                .takes_value(false)
```
Comment (CriesofCarrots) [inline banking-bench/src/main.rs:219]: This seems broken...? How can `log_messages_bytes_limit` end up anything other than None for banking-bench?

File: validator/src/main.rs [hunk 2]
```diff
@@ -2637,6 +2645,7 @@ pub fn main() {
         accounts_shrink_ratio,
         runtime_config: RuntimeConfig {
             bpf_jit: !matches.is_present("no_bpf_jit"),
+            log_messages_bytes_limit: value_t!(matches, "log_messages_bytes_limit", usize).ok(),
```
Comment (CriesofCarrots) [inline validator/src/main.rs:2648]: I think `value_of(&matches, "log_messages_bytes_limit")` should work, and would be a simpler way to write this.

Comment (willhickey) [reply validator/src/main.rs:2648]: Opened PR #26644 to cleanup these issues.

**Pull request merged**